sdk/python/feast/feature_store.py

```diff
@@ -333,17 +333,23 @@ def materialize(
     def _materialize_single_feature_view(
         self, feature_view: FeatureView, start_date: datetime, end_date: datetime
     ) -> None:
+        join_keys = []
+        registry = self._get_registry()
+        for entity_name in feature_view.entities:
+            entity = registry.get_entity(entity_name, self.config.project)
+            join_keys.append(entity.join_key)
+
         (
-            entity_names,
+            join_key_columns,
```

Comment:
If you are going to rename entity_names to join_key_columns then you should maybe make feature_names feature_name_columns